### PR TITLE
feat(Usage Reports): add billing snapshot config service to usage reports

### PR DESCRIPTION
**Pull request by manukm-ibm**
## PR summary
Introducing new CRUD APIs to configure billing reports snapshot.
TRI: https://github.ibm.com/BSS/architecture/blob/master/design-docs/services/billing/apptio-integration.md
Issue: https://github.ibm.com/BSS/metering-deliverables/issues/1990

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

File: usagereportsv4/usage_reports_v4.go [hunk 1]
```diff
@@ -1029,6 +1512,90 @@ func (options *GetOrgUsageOptions) SetHeaders(param map[string]string) *GetOrgUs
...
+// GetReportsSnapshotOptions : The GetReportsSnapshot options.
+type GetReportsSnapshotOptions struct {
+	// Account ID for which the billing report snapshot is requested.
+	AccountID *string `json:"account_id" validate:"required"`
+
+	// The month for which billing report snapshot is requested.  Format is yyyy-mm.
+	Month *string `json:"month" validate:"required"`
+
+	// Timestamp in milliseconds for which billing report snapshot is requested.
+	DateFrom *float64 `json:"date_from,omitempty"`
```
Comment (padamstx) [inline usagereportsv4/usage_reports_v4.go:1552]: this should probably be of type `*int64`.  Are you sure that the API definition is correct and agrees with the server implementation?

File: usagereportsv4/usage_reports_v4.go [hunk 2]
```diff
@@ -2203,132 +2812,736 @@ func UnmarshalResourcesSummary(m map[string]json.RawMessage, result interface{})
...
+	CreatedOn *string `json:"created_on,omitempty"`
+
+	// List of report types configured for the snapshot.
+	ReportTypes []SnapshotListSnapshotsItemReportTypesItem `json:"report_types,omitempty"`
+
+	// List of location of reports.
+	Files []SnapshotListSnapshotsItemFilesItem `json:"files,omitempty"`
+
+	// Timestamp at which snapshot is captured.
+	ProcessedAt *float64 `json:"processed_at,omitempty"`
```
Comment (padamstx) [inline usagereportsv4/usage_reports_v4.go:3082]: Another timestamp type property that is of type float64.

File: usagereportsv4/usage_reports_v4.go [hunk 3]
```diff
@@ -2203,132 +2812,736 @@ func UnmarshalResourcesSummary(m map[string]json.RawMessage, result interface{})
...
+	CosBucket *string `json:"cos_bucket,omitempty"`
+
+	// Region of the COS instance.
+	CosLocation *string `json:"cos_location,omitempty"`
+
+	// The endpoint of the COS instance.
+	CosEndpoint *string `json:"cos_endpoint,omitempty"`
+
+	// Timestamp in milliseconds when the snapshot configuration was created.
+	CreatedAt *float64 `json:"created_at,omitempty"`
```
Comment (padamstx) [inline usagereportsv4/usage_reports_v4.go:3309]: another timestamp that is a float64 value

File: usagereportsv4/usage_reports_v4_integration_test.go [hunk 4]
```diff
@@ -118,22 +96,15 @@ var _ = Describe(`UsageReportsV4 Integration Tests`, func() {
 			shouldSkipTest()
 		})
 		It(`GetAccountSummary(getAccountSummaryOptions *GetAccountSummaryOptions)`, func() {
-
 			getAccountSummaryOptions := &usagereportsv4.GetAccountSummaryOptions{
-				AccountID:    &accountID,
-				Billingmonth: &billingMonth,
+				AccountID: core.StringPtr("testString"),
```
Comment (padamstx) [inline usagereportsv4/usage_reports_v4_integration_test.go:100]: You have re-generated the integration tests and overwritten the code that was there previously, including manually developed code.  Is this really what you want?
You can see from this set of diffs that this test block previously used `accountId` and `billingMonth` which are variables that are obtained from the .env configuration file.

**Review by padamstx — CHANGES_REQUESTED** [review]
There are a few questions/issues to be addressed:
1. Did you really intend to completely overwrite the previous version of the integration tests?   There was manually developed code there that probably should be retained.  Please see https://github.com/IBM/platform-services-go-sdk/blob/main/update_service.md#updating-integration-testsexamples for instructions on how to update the integration tests and examples code while still retaining the previous manual changes.
2. There were no changes to the examples code, but yet the updates to the generated service code show 5 new operations.
3. It looks like a handful of timestamp-related properties that are supposed to hold "millisecond" values were defined in the API definition as "type number", which results in a language-specific floating point type being used in the generated code.   Are you sure that the server behavior supports a floating point value for these properties?   I would expect the API definition to use "type integer" and for the fields to be defined with a language-specific integer type.

**Comment by padamstx** [comment]
@manukm-ibm I took a fresh look at the changes and it looks like 6 operations have been added to the API (based on the changes to the generated service code):
- CreateReportsSnapshotConfig
- GetReportsSnapshotConfig
- UpdateReportsSnapshotConfig
- DeleteReportsSnapshotConfig
- GetReportsSnapshot

We'll need integration tests and working examples for each of those new operations.  It might help to generate the int tests and examples off to the side (i.e. into a different output directory), then copy/paste individual sections for each new operation into the checked-in version of the int tests and examples.
Reference: https://github.com/IBM/platform-services-go-sdk/blob/main/update_service.md#updating-integration-testsexamples

File: go.mod [hunk 5]
```diff
@@ -1,4 +1,4 @@
-module github.com/IBM/platform-services-go-sdk
+module github.com/manukm-ibm/platform-services-go-sdk
```
Comment (padamstx) [inline go.mod:1]: Please revert this change.  It looks like it's causing errors in the build.

**Review by padamstx — CHANGES_REQUESTED** [review]

File: go.mod (hunk 5 above)
Comment (manukm-ibm) [reply go.mod]: Done. This was a temporary change to test my TF scripts

File: usagereportsv4/usage_reports_v4_integration_test.go [hunk 6]
```diff
@@ -355,4 +373,108 @@ var _ = Describe(`UsageReportsV4 Integration Tests`, func() {
...
 		})
 	})
+
+	Describe(`CreateReportsSnapshotConfig - Setup the snapshot configuration`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+		It(`CreateReportsSnapshotConfig(createReportsSnapshotConfigOptions *CreateReportsSnapshotConfigOptions)`, func() {
+			fmt.Printf("TEST VALUES: %s", cosBucket)
+			fmt.Printf("TEST VALUES:%s", cosLocation)
```
Comment (padamstx) [inline usagereportsv4/usage_reports_v4_integration_test.go:383]:
```suggestion
			fmt.Fprintf(GinkgoWriter, "COS Bucket: %s", cosBucket)
			fmt.Fprintf(GinkgoWriter, "COS Location:%s", cosLocation)
```
Please use `fmt.Fprintf(GinkgoWriter,...)` instead of `fmt.Printf(...)`.  This will ensure that output is displayed only when the `-ginkgo.v` command line option is used.
You can find other instances of this in the same source file.

**Review by padamstx — CHANGES_REQUESTED** [review]
Looks good, but I had just one minor suggested change in the integration tests.

**Comment by padamstx** [comment]
@manukm-ibm For some reason, it looks like this PR contains 28 commits, but most of those are commits that already were merged into main, so this looks pretty strange :)   Could you perhaps rebase your feature branch on the current main branch from the primary repo and clear that up?

File: usagereportsv4/usage_reports_v4_integration_test.go (hunk 6 above)
Comment (manukm-ibm) [reply usagereportsv4/usage_reports_v4_integration_test.go:383]: Oh this was a printf for testing. Will remove it.

**Comment by manukm-ibm** [comment]
Integration test results of new changes
Screenshot 2023-09-14 at 11 59 49 AM

Examples results of new changes
Screenshot 2023-09-14 at 12 44 16 PM

**Review by padamstx — APPROVED** [review]
LGTM
